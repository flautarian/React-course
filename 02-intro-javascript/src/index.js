console.log("Hello world!!!")

const dice = {
    faces: 20,
    getRandomValue: function(){
        // 👇️ get number between min (inclusive) and max (inclusive)
        return Math.floor(Math.random() * (this.faces+ 1));
    }
}

const diceResult =`${dice.getRandomValue()}`;
console.log(diceResult);