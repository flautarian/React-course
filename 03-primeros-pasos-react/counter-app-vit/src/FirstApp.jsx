import React from 'react'
import { Fragment } from 'react';

function getPerson() {
    let name = prompt("cual es tu nombre?") || "leonidas";
    let surname = prompt("cual es tu apellido?") || "deuteronomus";
    let job = prompt("cual es tu oficio?") || "espartano";
    return {
        name: name,
        surname: surname,
        job: job
    };
}

let person = getPerson();

export const FirstApp = () => {
    return (
        <Fragment>
            <p>Nombre:</p>
            <p>-Mi nombre es {person.name}</p>
            <p>-Mi apellido es {person.surname}</p>
            <p>-Y mi trabajo es {person.job}</p>
        </Fragment>
    )
}
