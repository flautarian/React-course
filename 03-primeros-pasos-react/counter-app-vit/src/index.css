html, body {
    background-color: #21232A;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1em;
    padding: 70px;
  }
  
  code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
      monospace;
  }
  
button {
  padding: 15px;
  margin: 5px;
  font-size: 1.3em;
}