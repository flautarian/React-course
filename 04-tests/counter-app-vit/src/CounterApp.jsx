import React from 'react'
import { Fragment } from 'react'
import PropTypes from 'prop-types';
import { useState } from 'react';


export const CounterApp = ({value}) => {

    const [ counter, setCounter ] = useState( value );

    const addOne = ( event ) => setCounter((c) => c + 1 );

    const substractOne = ( event ) => setCounter((c) => c - 1 );

    const resetValue = ( event ) => setCounter((c) => c = value);
    
    
    return (
        <Fragment>
            <h1>CounterApp</h1>
            <h2>Valor: { counter } </h2>
            <button onClick={ (event) => addOne( event ) }>+1</button>
            <button onClick={ (event) => substractOne( event ) }>-1</button>
            <button onClick={ (event) => resetValue( event ) }>Reset</button>
        </Fragment>
    )
}

CounterApp.propTypes ={
    value: PropTypes.number.isRequired
}