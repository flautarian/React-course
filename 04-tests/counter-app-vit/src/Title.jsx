import React from 'react';
import { Fragment } from 'react';
import PropTypes from 'prop-types';

export const Title = ({ title, subtitle }) => {

  if(!title) throw new Error('El titulo es obligatorio');
  if(!subtitle) throw new Error('El subtitulo es obligatorio');

  return (
    <Fragment>
      <h1 data-testid="test-title">{title}</h1>
      <h2>{subtitle}</h2>
    </Fragment>
  )
}


Title.propTypes = {
  subtitle: PropTypes.string,
  title: PropTypes.string.isRequired
}

Title.defaultProps= {
  subtitle: "subtitulo no disponible",
  /* title: "titulo no disponible" */
}