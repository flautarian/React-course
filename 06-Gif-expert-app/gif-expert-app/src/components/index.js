export * from  './add-category'
export * from  './gif-expert-categories-grid'
export * from  './gif-expert-item'
