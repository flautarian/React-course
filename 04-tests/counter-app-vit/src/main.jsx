import React from "react";
import ReactDOM from "react-dom/client";
import { Title } from './Title';
import { FirstApp } from "./FirstApp";
import './index.css';
import { CounterApp } from "./CounterApp";

let valueCounterApp = 10;

ReactDOM.createRoot( document.getElementById('root')).render(
  <React.StrictMode>
    <Title title="Hola!, bienvenido a mi App" subtitle="subtitulo"/>
    <FirstApp/>
    <CounterApp value={valueCounterApp}/>
  </React.StrictMode>
)