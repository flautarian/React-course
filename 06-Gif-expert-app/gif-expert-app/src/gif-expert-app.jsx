import { useState, Fragment } from "react";
import { AddCategory, GifExpertCategoriesGrid } from "./components/index";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'One piece']);

    const onAddCategory = (value) => {
        console.log(value);
        if (!!value && !categories.includes(value) )
            setCategories([value, ...categories]);
    }

    return (
        <Fragment>
            {/* Title */}
            <h1>Gif Expert App</h1>
            {/* Input */}
            <AddCategory onAddCategory={onAddCategory} />
            {/* Results */}
            {categories.map(category => {
                return (
                    <GifExpertCategoriesGrid key={category} category={ category }/>
                );
            })}
            
            
        </Fragment>
    )
}