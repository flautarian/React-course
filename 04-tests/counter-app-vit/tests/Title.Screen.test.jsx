import { render, screen } from '@testing-library/react'
import { Title } from '../src/Title';

describe('Title screen component tests', () => {

    const testTitle= "title";
    const testSubtitle= "subtitle";
    
    test('should match snapshot', () => {
        const { container } = render( <Title title={ testTitle }/> );
        expect( container ).toMatchSnapshot();
    });

    test('should count 1 subtitle-test-element', () => {
        render( <Title title={ testTitle } subtitle={ testSubtitle }/> );
        expect( screen.getAllByText(testSubtitle).length ).toBe(1);
    });

});