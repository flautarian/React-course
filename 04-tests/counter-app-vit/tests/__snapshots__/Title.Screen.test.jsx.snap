// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component tests should match screen elements 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    title
  </h1>
  <h2>
    subtitulo no disponible
  </h2>
</div>
`;

exports[`Title component tests should match snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    title
  </h1>
  <h2>
    subtitulo no disponible
  </h2>
</div>
`;

exports[`Title component tests should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    title
  </h1>
  <h2>
    subtitulo no disponible
  </h2>
</div>
`;

exports[`Title component tests should show test-element 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    title
  </h1>
  <h2>
    subtitulo no disponible
  </h2>
</div>
`;

exports[`Title screen component tests should match snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    title
  </h1>
  <h2>
    subtitulo no disponible
  </h2>
</div>
`;
